sklearn/base.py

```diff
@@ -50,11 +50,11 @@ def clone(estimator, *, safe=True):
 
     Parameters
     ----------
-    estimator : {list, tuple, set} of estimator objects or estimator object
+    estimator : {list, tuple, set} of BaseEstimator or a single BaseEstimator
```

Comment:
Actually it should be an object inheriting from `BaseEstimator`. From what I recall, we are using `estimator instance` more and more (cf. `plot_roc_curve`).

I think that this rule, whatever the one we choose should be added to the contributing guide to avoid confusion indeed.